src/main/java/org/skriptlang/skript/bukkit/recipes/elements/SecRegisterRecipe.java

```diff
@@ -0,0 +1,272 @@
+		shapedRecipe.shape("abc","def","ghi");
+		for (int i = 0; i < ingredients.length; i++) {
+			RecipeChoice thisChoice = ingredients[i];
+			if (thisChoice != null)
+				shapedRecipe.setIngredient(characters[i], thisChoice);
+		}
+		completeRecipe(key, shapedRecipe);
+	}
+
+	private void createShapelessRecipe(NamespacedKey key, ItemStack result, RecipeChoice[] ingredients, String group, CraftingBookCategory category) {
+		ShapelessRecipe shapelessRecipe = new ShapelessRecipe(key, result);
+		if (category != null)
+			shapelessRecipe.setCategory(category);
+		if (group != null && !group.isEmpty())
+			shapelessRecipe.setGroup(group);
+		for (int i = 0; i < ingredients.length; i++) {
+			RecipeChoice thisChoice = ingredients[i];
+			if (thisChoice != null)
+				shapelessRecipe.addIngredient(thisChoice);
+		}
+		completeRecipe(key, shapelessRecipe);
+	}
+
+	private void createCookingRecipe(RecipeType recipeType, NamespacedKey key, ItemStack result, RecipeChoice input, String group, CookingBookCategory category, int cookingTime, float experience) {
+		var recipe = switch (recipeType) {
+			case BLASTING -> new BlastingRecipe(key, result, input, experience, cookingTime);
+			case CAMPFIRE -> new CampfireRecipe(key, result, input, experience, cookingTime);
+			case FURNACE -> new FurnaceRecipe(key, result, input, experience, cookingTime);
+			case SMOKING -> new SmokingRecipe(key, result, input, experience, cookingTime);
+			default -> throw new IllegalStateException("Unexpected value: " + recipeType);
+		};
+		if (category != null)
+			recipe.setCategory(category);
+		if (group != null && !group.isEmpty())
+			recipe.setGroup(group);
+		completeRecipe(key, recipe);
+	}
+
+	private void createSmithingRecipe(RecipeType recipeType, NamespacedKey key, ItemStack result, RecipeChoice base, RecipeChoice template, RecipeChoice addition) {
+		if (base == null || (template == null && recipeType != RecipeType.SMITHING) || addition == null) {
+			customError("Unable to create '" + recipeType + "' recipe, missing data.");
+			return;
+		}
+		var recipe = switch (recipeType) {
+			case SMITHING_TRANSFORM -> new SmithingTransformRecipe(key, result, template, base, addition);
+			case SMITHING_TRIM -> new SmithingTrimRecipe(key, template, base, addition);
+			case SMITHING -> new SmithingRecipe(key, result, base, addition);
+			default -> throw new IllegalStateException("Unexpected value: " + recipeType);
+		};
+		completeRecipe(key, recipe);
+	}
+
+	private void createStonecuttingRecipe(NamespacedKey key, ItemStack result, RecipeChoice input) {
+		if (input == null) {
+			customError("Unable to create a 'stonecutting' recipe, missing data.");
+			return;
+		}
+		StonecuttingRecipe recipe = new StonecuttingRecipe(key, result, input);
+		completeRecipe(key, recipe);
+	}
+
+	private void customError(String message) {
+		Skript.info("Line " + thisNode.getLine() + ": (" + thisScript + ")\n\t" + message + "\n\t" + thisNode.getKey());
+	}
```

Comment:
please either do this properly or don't do it at all
properly would entail an api that all syntaxes could use that uses proper lang nodes and colours like all the rest of the error formats.